AdvancedTeleport-Bukkit/src/main/java/io/github/niestrat99/advancedteleport/config/CustomMessages.java

```diff
@@ -1,246 +1,312 @@
 public final class CustomMessages extends ATConfig {
 
     public static CustomMessages config;
     private static HashMap<CommandSender, BukkitRunnable> titleManager;
 
+    @NotNull private static ImmutableMap<String, PartialComponent> messageCache = ImmutableMap.of();
+    @NotNull private static ImmutableSortedSet<String> prefixes = ImmutableSortedSet.of();
+
+    @Nullable private static BukkitAudiences audience;
+
     public CustomMessages() throws IOException {
         super("custom-messages.yml");
         config = this;
         titleManager = new HashMap<>();
+        populate();
+
+        if (!PaperLib.isPaper()) {
+            audience = BukkitAudiences.create(CoreClass.getInstance());
+        }
+    }
+
+    @Override
+    public void reload() throws IOException {
+        super.reload();
+        populate();
     }
 
+    @Override
     public void loadDefaults() {
+        makeSectionLenient("Common");
+        addDefault(
+            "Common.prefixes",
+            List.of("<aqua>↑</aqua> <dark_gray>»</dark_gray>"),
+            """
+            The prefixes for messages, the first element of this list will be usable as <prefix>,
+            with each element after that being usable as <prefix:index> with index being the items index in the list.
+            """.trim()
+        );
+
         makeSectionLenient("Teleport");
-        addDefault("Teleport.eventBeforeTP" , "&b↑ &8» &7Teleporting in &b{countdown} seconds&7, please do not move!");
+        addDefault("Teleport.eventBeforeTP" , "<prefix> <gray>Teleporting in <aqua><countdown> seconds</aqua>, please do not move!");
 
         addComment("Teleport.eventBeforeTP_title", "This is an example use for titles and subtitles in the plugin." +
                 "\nThis feature is supported only if you're on version 1.8.8 or newer.");
         addExample("Teleport.eventBeforeTP_title.length" , 80 , "How many seconds (in ticks) the header should last. This is not including fading.");
         addExample("Teleport.eventBeforeTP_title.fade-in" , 0 , "How many seconds (in ticks) the header should take to fade in.");
         addExample("Teleport.eventBeforeTP_title.fade-out" , 10 , "How many seconds (in ticks) the header should take to fade out.");
-        addExample("Teleport.eventBeforeTP_title.0", "&7&lTeleporting...");
-        addExample("Teleport.eventBeforeTP_title.20", "&b> &7&lTeleporting... &b<");
-        addExample("Teleport.eventBeforeTP_title.40", "&b>> &7&lTeleporting... &b<<");
+        addExample("Teleport.eventBeforeTP_title.0", "<gray><bold>Teleporting...");
+        addExample("Teleport.eventBeforeTP_title.20", "<aqua>></aqua>&lTeleporting... &b<");
+        addExample("Teleport.eventBeforeTP_title.40", "<aqua>>></aqua>&lTeleporting... &b<<");
         addExample("Teleport.eventBeforeTP_title.60", "&b>>> &e&lTeleported! &b<<<");
         addExample("Teleport.eventBeforeTP_subtitle.0", "&bPlease do not move!");
```

Comment:
Some of these aren't changed to MM format (e.g. &l to `<b>`), is there a reason for that?